BMSForCharging.h

```diff
@@ -0,0 +1,26 @@
+#include "BatteryManagementSystem.h"
+class BatteryMgmtSystmForCharging : public BatteryManagemtSystem
+{
+public:
+    void sendAlert(){
+        //stop charging and Beep sound
+    }
+    bool monitorTemparature(){
+        while(getTemperature() >= TEMPERATURE_LOWER_LIMIT && getTemperature() <= TEMPERATURE_UPPER_LIMIT);

```

Comment:
this while loop will run continuously and use CPU. One way to stop this method from consuming CPU would be to trigger the check whenever temperature is modified or value changed.